diff --git a/glib/CMakeLists.txt b/glib/CMakeLists.txt
index a368eded..6d82ed07 100644
--- a/glib/CMakeLists.txt
+++ b/glib/CMakeLists.txt
@@ -40,11 +40,13 @@ set(poppler_glib_public_headers
 )
 
 find_program(GLIB2_MKENUMS glib-mkenums)
+find_program(GLIB2_MKENUMS_PYTHON NAMES python3 python)
 
 add_custom_command(
   OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/poppler-enums.h
-  COMMAND ${GLIB2_MKENUMS}
+  COMMAND ${GLIB2_MKENUMS_PYTHON}
   ARGS
+    ${GLIB2_MKENUMS}
     --template poppler-enums.h.template
     ${poppler_glib_public_headers} > ${CMAKE_CURRENT_BINARY_DIR}/poppler-enums.h
   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
@@ -54,8 +56,9 @@ add_custom_command(
 
 add_custom_command(
   OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/poppler-enums.c
-  COMMAND ${GLIB2_MKENUMS}
+  COMMAND ${GLIB2_MKENUMS_PYTHON}
   ARGS
+    ${GLIB2_MKENUMS}
     --template poppler-enums.c.template
     ${poppler_glib_public_headers} > ${CMAKE_CURRENT_BINARY_DIR}/poppler-enums.c
   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
