diff --git a/fontforge/CMakeLists.txt b/fontforge/CMakeLists.txt
index f4d0e2cc4..477e95d76 100644
--- a/fontforge/CMakeLists.txt
+++ b/fontforge/CMakeLists.txt
@@ -237,6 +237,8 @@ target_link_libraries(fontforge
     ZLIB::ZLIB
 )
 
+set_target_properties(fontforge PROPERTIES PUBLIC_HEADER "${FONTFORGE_INST_HEADERS}")
+
 if(APPLE)
   target_link_libraries(fontforge
     PRIVATE
@@ -269,11 +271,8 @@ else()
   target_link_libraries(fontforge PRIVATE gunicode gutils)
 endif()
 
-# No dev package -> no need to install if static
-if(BUILD_SHARED_LIBS)
-  if(WIN32)
-    install(TARGETS fontforge RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-  else()
-    install(TARGETS fontforge RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
-  endif()
+if(WIN32 OR CYGWIN)
+  install(TARGETS fontforge RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/fontforge)
+else()
+  install(TARGETS fontforge RUNTIME DESTINATION ${CMAKE_INSTALL_LIBDIR} LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/fontforge)
 endif()
diff --git a/inc/CMakeLists.txt b/inc/CMakeLists.txt
index b40a6da24..23372b90a 100644
--- a/inc/CMakeLists.txt
+++ b/inc/CMakeLists.txt
@@ -23,3 +23,32 @@ target_link_libraries(fontforge_common_headers
     Gcov::Gcov
 )
 endif()
+
+set(FONTFORGE_INST_COMMON_HEADERS
+  basics.h
+  carbon.h
+  chardata.h
+  charset.h
+  dlist.h
+  ffglib.h
+  "${CMAKE_CURRENT_BINARY_DIR}/fontforge-config.h"
+  gdraw.h
+  gfile.h
+  ggadget.h
+  gicons.h
+  gimage.h
+  gio.h
+  gkeysym.h
+  gprogress.h
+  gresedit.h
+  gresource.h
+  gwidget.h
+  gwwiconv.h
+  intl.h
+  ustring.h
+  utype.h
+)
+
+set_target_properties(fontforge_common_headers PROPERTIES PUBLIC_HEADER "${FONTFORGE_INST_COMMON_HEADERS}")
+
+install(TARGETS fontforge_common_headers PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
